components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceManagementServiceImpl.java

```diff
@@ -471,6 +472,33 @@ public Response getDeviceLocation(
 
     }
 
+
+    @GET
+    @Path("/{type}/{id}/info")
+    @Override
+    public Response getDeviceInformation(
+            @PathParam("type") @Size(max = 45) String type,
+            @PathParam("id") @Size(max = 45) String id,
+            @HeaderParam("If-Modified-Since") String ifModifiedSince) {
+        DeviceInformationManager informationManager;
+        DeviceInfo deviceInfo;
+        try {
+            DeviceIdentifier deviceIdentifier = new DeviceIdentifier();
+            deviceIdentifier.setId(id);
+            deviceIdentifier.setType(type);
+            informationManager = DeviceMgtAPIUtils.getDeviceInformationManagerService();
+            deviceInfo = informationManager.getDeviceInfo(deviceIdentifier);
+
+        } catch (DeviceDetailsMgtException e) {
+            String msg = "Error occurred while getting the device information.";
```

Comment:
Add deviceId and type as parameters to the msg. That would give enough context for logs.

Reply:
Fixed in https://github.com/wso2/carbon-device-mgt/pull/965/commits/a7e41faa19ecca47fa1356f259f59d771194484a